autoPyTorch/pipeline/components/setup/network_backbone/ResNetBackbone.py

```diff
@@ -327,8 +320,7 @@ def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
             # if in_features != out_features
             # -> result = W_shortcut(A(BN(x))) + W_2(~D(A(BN(W_1(A(BN(x))))))
             x = self.start_norm(x)
-            if self.config["use_skip_connection"]:
-                residual = self.shortcut(x)
+            residual = self.shortcut(x)
```

Comment:
From here to until the return, I would suggest the following change:
```suggestion
        if not self.config["use_skip_connection"]:
            # Early-return
            return self.layers(x)

        # We use skip connection
        residual = self.shortcut(x)

        if self.config["multi_branch_choice"] == 'shake-shake':
            x1 = self.layers(x)
            x2 = self.shake_shake_layers(x)
            alpha, beta = shake_get_alpha_beta(self.training, x.is_cuda)
            x = shake_shake(x1, x2, alpha, beta)
        elif self.config["multi_branch_choice"] == 'shake-drop':
            x = self.layers(x)
            alpha, beta = shake_get_alpha_beta(self.training, x.is_cuda)
            bl = shake_drop_get_bl(
                self.block_index,
                1 - self.config["max_shake_drop_probability"],
                self.num_blocks,
                self.training,
                x.is_cuda,
            )
            x = shake_drop(x, alpha, beta, bl)
        else:
            x = self.layers(x)

        return x + residual
```

Reply:
To check

Reply:
I think your code has a small problem if I validated it correctly, it is the case where there is a skip connection and there is no need for a shortcut to match the dimension change, so when the skip connection dimensions have not changed. And I think I addressed it, other than that it is better than was what before.

Reply:
Thanks, I did not check the process before the lines!